packages/override-tools/src/GitReactFileRepository.ts

```diff
@@ -81,6 +81,7 @@ export default class GitReactFileRepository
           `refs/tags/v${reactNativeVersion}`,
         );
         await this.gitClient.checkout('FETCH_HEAD');
+        await this.gitClient.reset('hard');
```

Comment:
During testing I discovered my local temp Git repo somehow got into a bad state, where certain files were deleted. I'm not sure how, but adding this should ensure that shouldn't happen.